src/main/java/com/epam/task/purchases/DiscountIfMorePurchase.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.task.purchases;
+
+
+public class DiscountIfMorePurchase extends AbstractPurchase {
+    private static final double QUANTITY_TO_DISCOUNT = 3;
+    private final double discountPercentage;
+
+    public DiscountIfMorePurchase(Product product, int purchasedNum, double discountPercentage) {
+        super(product, purchasedNum);
+        this.discountPercentage = discountPercentage;
+    }
+
+    @Override
+    protected Euro getFinalCost(Euro baseCost) {
+        if (purchasedNum >= QUANTITY_TO_DISCOUNT){
+            return baseCost.mul(1 - discountPercentage/100, RoundMethods.ROUND, 0);
```

Comment:
```suggestion
            baseCost = baseCost.mul(1 - discountPercentage/100, RoundMethods.ROUND, 0);
```
i don't insist, just my preference: the only `return` if a method is non `boolean`.